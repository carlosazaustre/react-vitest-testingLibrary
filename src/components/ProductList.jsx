export default function ProductList({items}) {
	return (
        <ul>
            {items.map((item) => (
                <li key={item.id} value={item.name}>{item.name} - ${item.price}</li>
            ))}
        </ul>
	);
}