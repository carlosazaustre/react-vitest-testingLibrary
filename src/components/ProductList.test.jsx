import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import ProductList from './ProductList';

const mockProducts = [
    {
        id: 1,
        name: "Cachopo",
        price: 30,
    },
    {
        id: 2,
        name: "Chorizo a la sidra",
        price: 15,
    },
    {
        id: 3,
        name: "Navajas",
        price: 25,
    },
];

describe('<ProductList />', () => {
    it('Should render a list of products', () => {
        render(<ProductList items={mockProducts} />);
        const products = screen.getAllByRole('listitem');
        
        expect(products).toHaveLength(3);
    });

    it('should match snapshot', () => {
        render(<ProductList items={mockProducts} />);
        expect(screen.getByRole('list')).toMatchSnapshot();
    });
});