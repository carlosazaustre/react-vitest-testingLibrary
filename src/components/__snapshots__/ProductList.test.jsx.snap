// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ProductList /> > should match snapshot 1`] = `
<ul>
  <li
    value="Cachopo"
  >
    Cachopo
     - $
    30
  </li>
  <li
    value="Chorizo a la sidra"
  >
    Chorizo a la sidra
     - $
    15
  </li>
  <li
    value="Navajas"
  >
    Navajas
     - $
    25
  </li>
</ul>
`;
